marxs/optics/multiLayerMirror.py

```diff
@@ -92,14 +93,17 @@ def process_photons(self, photons):
         spread_refl = np.interp(local_intersection[:,1], local_coords_in_file, reflectFile['FWHM(nm)'])
         
         wavelength = 1.23984282 / photons['energy']   # wavelength is in nm assuming energy is in keV
-        c_squared = (spread_refl ** 2) / (2 * np.log(2))
-        refl_prob = max_refl * np.exp(-((wavelength - peak_wavelength) ** 2) / (2 * c_squared))
+        c_squared = (spread_refl ** 2) / (8. * np.log(2))
+        c_is_zero = (c_squared == 0)
+    	
+        refl_prob = np.zeros(len(wavelength))
+        refl_prob[~c_is_zero] = max_refl[~c_is_zero] * np.exp(-((wavelength[~c_is_zero] - peak_wavelength[~c_is_zero]) ** 2) / (2 * c_squared[~c_is_zero]))
         
         # find probability of being reflected due to polarization
         # v_1 is parallel to plane, good reflection direction
-        refl_prob *= np.einsum('ij,ij->i', polarization, v_1)**2
+        refl_prob[~c_is_zero] *= np.einsum('ij,ij->i', polarization[~c_is_zero], v_1[~c_is_zero]) ** 2
         refl_prob[np.isnan(refl_prob)] = 0
         # multiply probability by probability of reflection
         photons['probability'] *= refl_prob / 100
         
-        return photons
+        return photons[photons['probability'] != 0]
```

Comment:
So far, I've always taken the approach to return all photons, even those with probability 0. I know that your docstring at the top says "will be removed" and I think that's fine, too.
I'm just thinking that leaving them in allows to do things like `len(photons[probability >0]) / len(photons)` which is the fraction of photons that make it. 
Let me know what you think and then we'll merge.